pkg/clustermesh/clustermesh_test.go

```diff
@@ -138,15 +141,31 @@ func (s *ClusterMeshTestSuite) TestClusterMesh(c *C) {
 		Context: ctx,
 	})
 	defer ipc.Shutdown()
-	cm, err := NewClusterMesh(Configuration{
-		Name:                  "test2",
-		ConfigDirectory:       dir,
-		NodeKeyCreator:        testNodeCreator,
-		nodeObserver:          &testObserver{},
-		RemoteIdentityWatcher: mgr,
-		IPCache:               ipc,
-	})
-	c.Assert(err, IsNil)
+
+	var cm *ClusterMesh
+	hive := hive.New(
```

Comment:
FYI you could also call `NewClusterMesh` directly and use `hivetest.Lifecycle`.

Reply:
Example: https://github.com/cilium/cilium/blob/c8598f8227665054d05cfeec3f65b15005f087c5/pkg/maps/egressmap/policy_test.go#L26 Saves you from creating a hive.

Reply:
Cool, I didn't know about that. Let me make a try.